### PR TITLE
Add `UserSettings` entity

### LINKED ISSUE #214: Add `UserSettings` entity
### Description
Add `UserSettings` entity

### Requirements
- project should be runnable
- relations must be correct

### Deliverables
- entity configuration
- repository
- services
- mappers
- validations

### PR DESCRIPTION
**Pull request by abdura52**
Closes #214 

User settings repository & interface
user settings service  and interface
User settings  validator
userDto model
Mapper
Notification type enum
Theme enum

added

### CODE REVIEW

File: src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Enums/NotificationType.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿namespace AirBnB.Domain.Enums;
+
+public enum NotificationType
+{
```
Comment (botirjonabdumannonovv) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Enums/NotificationType.cs:4]: NotificationType enumga ham comment yozish kerak edi

File: src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Enums/Theme.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿namespace AirBnB.Domain.Enums;
+
+public enum Theme
+{
```
Comment (botirjonabdumannonovv) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Enums/Theme.cs:4]: Theme enumga ham comment yozish kerak edi

**Review by botirjonabdumannonovv — DISMISSED** [review]

**Review by botirjonabdumannonovv — APPROVED** [review]

File: src/Training.AirBnb.Clone.Backend/AirBnB.Application/Common/Identity/Services/IUserSettingsService.cs [hunk 3]
```diff
@@ -0,0 +1,92 @@
+﻿using AirBnB.Domain.Common.Query;
+using AirBnB.Domain.Entities;
+using System.Linq.Expressions;
+
+namespace AirBnB.Application.Common.Identity.Services;
+
+public interface IUserSettingsService
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Application/Common/Identity/Services/IUserSettingsService.cs:7]: Add comment as "Defines user settings foundation service functionalities"

File: src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Enums/Theme.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    /// <summary>
+    /// Dark theme
+    /// </summary>
+    Dark = 0,
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Enums/Theme.cs:8]: Add `SystemTheme` as first value, and `Dark` as last value

File: src/Training.AirBnb.Clone.Backend/AirBnB.Infrastructure/Common/Validators/UserSettingsValidator.cs [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+/// <summary>
+/// Validator class for validating user settings data using fluent validation
+/// </summary>
+public class UserSettingsValidator : AbstractValidator<UserSettings>
+{
+    public UserSettingsValidator()
+    {
+        RuleFor(userSettings => userSettings.Id).NotEqual(Guid.Empty);
+        RuleFor(userSettings => userSettings.PreferredTheme).NotEmpty().NotNull();
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Infrastructure/Common/Validators/UserSettingsValidator.cs:14]: Remove this

File: src/Training.AirBnb.Clone.Backend/AirBnB.Persistence/EntityConfigurations/UserSettingsConfiguration.cs [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+public class UserSettingsConfiguration : IEntityTypeConfiguration<UserSettings>
+{
+    public void Configure(EntityTypeBuilder<UserSettings> builder)
+    {
+        builder
+            .HasOne(userSettings => userSettings.User)
+            .WithOne(user => user.UserSettings)
+            .HasForeignKey(typeof(UserSettings), "UserId");
+    }
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Persistence/EntityConfigurations/UserSettingsConfiguration.cs:15]: Add as `HasForeignKey<UserSettings>`

**Review by sulton-max — CHANGES_REQUESTED** [review]

**Review by botirjonabdumannonovv — APPROVED** [review]

**Comment by botirjonabdumannonovv** [comment]
Descriptionga Closes deb branchizni yozib qo'ying

**Review by Habiba67y — APPROVED** [review]

**Review by sulton-max — APPROVED** [review]
Good job

**Review by Habiba67y — APPROVED** [review]

**Pull request merged**